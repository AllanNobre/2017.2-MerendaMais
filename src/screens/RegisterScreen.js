import React from 'react';
import PropTypes from 'prop-types';
import { Text, ScrollView, View, TextInput, TouchableOpacity, Picker } from 'react-native';
import { Actions } from 'react-native-router-flux';

const styles = {

  principal: {
    backgroundColor: 'white',
    flex: 1,
  },

  topo: {
    flex: 1.2,
    paddingTop: 10,
    paddingBottom: 10,
    backgroundColor: '#FF9500',
    borderBottomColor: 'black',
    borderBottomWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },

  textLogo: {
    fontSize: 35,
    color: 'white',
    fontWeight: 'bold',
    marginTop: 10,
  },

  rodape: {
    flex: 0.7,
    borderTopColor: '#a9a9a9',
    borderTopWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
  },

  InputStyle: {
    padding: 10,
    backgroundColor: '#FAFAFA',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: 'gray',
    borderWidth: 1,
    borderRadius: 7,
    marginHorizontal: 15,
    marginBottom: 10,
  },

  InputDropdown: {
    borderColor: 'gray',
    borderWidth: 1,
    borderRadius: 7,
    marginHorizontal: 15,
    marginBottom: 10,
  },

  buttonContainer: {
    paddingVertical: 10,
    borderWidth: 1,
    borderRadius: 7,
    marginHorizontal: 15,
    marginTop: 20,
    marginBottom: 20,
    backgroundColor: '#FF9500',
  },

  buttonText: {
    textAlign: 'center',
    color: '#FFF',
  },

};

export default class RegisterScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      cpf: '',
      name: '',
      email: '',
      phone: '',
      isPresident: '',
      password: '',
      segment: '',
      CAE_Type: '',
      CAE: '',
    };
  }

  render() {
    console.log(this.state);
    const password = this.state.isPresident === true ? (
      <View>
        <Text style={styles.container}>     Senha</Text>
        <TextInput
          placeholder="Digite sua senha"
          placeholderTextColor="#95a5a6"
          style={styles.InputStyle}
          underlineColorAndroid="transparent"
          returnKeyLabel={'next'}
          onChangeText={text => this.setState({ password: text })}
          secureTextEntry
        />
      </View>
    ) : null;

    return (
      <ScrollView style={styles.principal}>
        <View style={styles.topo}>
          <Text style={styles.textLogo}>Merenda +</Text>
        </View>

        <Text />
        <Text>     CPF</Text>
        <TextInput
          placeholder="Digite o seu CPF"
          placeholderTextColor="#95a5a6"
          style={styles.InputStyle}
          underlineColorAndroid="transparent"
          returnKeyLabel={'next'}
          onChangeText={text => this.setState({ cpf: text })}
        />
        <Text>     Nome</Text>
        <TextInput
          placeholder="Digite o seu nome completo"
          placeholderTextColor="#95a5a6"
          style={styles.InputStyle}
          underlineColorAndroid="transparent"
          returnKeyLabel={'next'}
          onChangeText={text => this.setState({ name: text })}
        />
        <Text>     Email</Text>
        <TextInput
          placeholder="Digite o seu email"
          placeholderTextColor="#95a5a6"
          style={styles.InputStyle}
          underlineColorAndroid="transparent"
          returnKeyLabel={'next'}
          onChangeText={text => this.setState({ email: text })}
        />
        <Text>     Telefone</Text>
        <TextInput
          placeholder="Digite o seu telefone"
          placeholderTextColor="#95a5a6"
          style={styles.InputStyle}
          underlineColorAndroid="transparent"
          returnKeyLabel={'next'}
          onChangeText={text => this.setState({ phone: text })}
        />
        <Text>     Cargo</Text>
        <View
          style={styles.InputDropdown}
        >
          <Picker
            onValueChange={value => this.setState({ isPresident: value, password: 'senha' })}
            selectedValue={this.state.isPresident}
          >
            <Picker.Item value="" label="Escolha seu cargo" color="#95a5a6" />
            <Picker.Item value label="Presidente" />
            <Picker.Item value={false} label="Conselheiro" />
          </Picker>
        </View>
        {password}
        <Text>     Segmento</Text>
        <View
          style={styles.InputDropdown}
        >
          <Picker
            onValueChange={value => this.setState({ segment: value })}
            selectedValue={this.state.segment}
          >
            <Picker.Item value="" label="Escolha seu segmento" color="#95a5a6" />
            <Picker.Item value="Suplente" label="Suplente" />
            <Picker.Item value="Titular" label="Titular" />
          </Picker>
        </View>
        <Text>     Tipo do CAE</Text>
        <TextInput
          placeholder="Escolha o tipo do seu CAE"
          placeholderTextColor="#95a5a6"
          style={styles.InputStyle}
          underlineColorAndroid="transparent"
          returnKeyLabel={'next'}
          onChangeText={text => this.setState({ CAE_Type: text })}
        />
        <Text>     CAE</Text>
        <TextInput
          placeholder="Lista com o CAE do seu município/estado"
          placeholderTextColor="#95a5a6"
          style={styles.InputStyle}
          underlineColorAndroid="transparent"
          returnKeyLabel={'next'}
          onChangeText={text => this.setState({ CAE: text })}
        />
        <TouchableOpacity
          onPress={() => this.props.createUser(this.state)}
          style={styles.buttonContainer}
          activeOpacity={0.7}
        >
          <Text style={styles.buttonText}>Concluir</Text>
        </TouchableOpacity>

        <View style={styles.rodape}>
          <TouchableOpacity
            onPress={() => Actions.loginConselheiro()}
            activeOpacity={0.6}
          >
            <Text>Já tem um cadastro?
              <Text style={{ color: 'blue' }}> Entrar</Text>
            </Text>
          </TouchableOpacity>
        </View>

      </ScrollView>
    );
  }
}

RegisterScreen.propTypes = {
  createUser: PropTypes.func.isRequired,

};
