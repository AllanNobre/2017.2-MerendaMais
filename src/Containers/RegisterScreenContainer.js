import { connect } from 'react-redux';
import { asyncRegisterCounselor } from '../actions/counselorActions';
import RegisterScreen from '../screens/RegisterScreen';

const mapStateToProps = state => (
  {
    message_erro: state.application.message_erro,
    isLoading: state.application.isLoading,
  }
);

const mapDispatchToProps = dispatch => ({
  asyncRegisterCounselor: userData => dispatch(asyncRegisterCounselor(userData)),
});

const RegisterScreenContainer = connect(mapStateToProps, mapDispatchToProps)(RegisterScreen);

export default RegisterScreenContainer;
