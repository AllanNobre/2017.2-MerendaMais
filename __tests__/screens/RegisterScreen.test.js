import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
// imported as a connected component!
import RegisterScreenContainer from '../../src/Containers/RegisterScreenContainer';
import RegisterScreen from '../../src/screens/RegisterScreen';


Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  counselor: {
    nuvemCode: 1,
    email: 'rodolfo@gmail.com',
    name: 'Rodolfo',
    userName: 'rodolfo@gmail.com',
    password: 'senha',
    token: 'tokenGenerico',
    profile: {
      cpf: '12312312312',
      phone: '96661234',
      isPresident: false,
      counselorType: 'Titular',
      segment: 'Poder executivo',
      CAE_Type: 'Estadual',
      CAE: 'Distrito Federal',
      CAE_UF: 'AC - Acre',
      CAE_municipalDistrict: 'Brazlândia',
    },
  },
  application: {
    isLoading: false,
    message_erro: '',
  },
};

const store = mockStore(initialState);

describe('Testing RegisterScreen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <RegisterScreenContainer />,
      { context: { store } },
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Testing RegisterScreen Input', () => {
  const wrapper = shallow(<RegisterScreen />);

  it('should change state when the text of cpf input component changes', () => {
    const cpfInputComponent = wrapper.find('TextInput').at(0);
    cpfInputComponent.simulate('ChangeText', '11111111111');
    expect(wrapper.state().profile.cpf).toEqual('11111111111');
  });

  it('should change state when the text of name input component changes', () => {
    const nameInputComponent = wrapper.find('TextInput').at(1);
    nameInputComponent.simulate('ChangeText', 'test');
    expect(wrapper.state('name')).toEqual('test');
  });

  it('should change state when the text of email input component changes', () => {
    const emailInputComponent = wrapper.find('TextInput').at(2);
    emailInputComponent.simulate('ChangeText', 'test5@test.com');
    expect(wrapper.state('email')).toEqual('test5@test.com');
  });

  it('should change state when the text of phone input component changes', () => {
    const phoneInputComponent = wrapper.find('TextInput').at(3);
    phoneInputComponent.simulate('ChangeText', '555555555');
    expect(wrapper.state().profile.phone).toEqual('555555555');
  });

  it('should change state when the text of isPresident input component changes', () => {
    const isPresidentInputComponent = wrapper.find('Picker').at(0);
    isPresidentInputComponent.simulate('valueChange', false);
    expect(wrapper.state().profile.isPresident).toEqual(false);
  });

  it('should change state when the text of counselorType input component changes', () => {
    const counselorTypeInputComponent = wrapper.find('Picker').at(1);
    counselorTypeInputComponent.simulate('valueChange', 'Suplente');
    expect(wrapper.state().profile.counselorType).toEqual('Suplente');
  });

  it('should change state when the text of segment input component changes', () => {
    const segmentInputComponent = wrapper.find('Picker').at(2);
    segmentInputComponent.simulate('valueChange', 'Pais de alunos');
    expect(wrapper.state().profile.segment).toEqual('Pais de alunos');
  });

  it('should change state when the text of CAE_Type input component changes', () => {
    const caeTypeInputComponent = wrapper.find('Picker').at(3);
    caeTypeInputComponent.simulate('valueChange', 'Estadual');
    expect(wrapper.state().profile.CAE_Type).toEqual('Estadual');
  });

  it('should change state when the text of UF input component changes', () => {
    const UFInputComponent = wrapper.find('Picker').at(4);
    UFInputComponent.simulate('valueChange', 'DF - Distrito Federal');
    expect(wrapper.state().profile.CAE_UF).toEqual('DF - Distrito Federal');
  });
});

describe('Testing RegisterScreen On pressed buttons', () => {
  it('Test if createUser button is pressed', () => {
    const asyncRegisterCounselor = (state) => {
      expect(state.nuvemCode).toEqual(1);
      expect(state.email).toEqual('Conselheiro@email.com');
      expect(state.name).toEqual('Conselheiro');
      expect(state.userName).toEqual('Conselheiro@email.com');
      expect(state.password).toEqual('senha');
      expect(state.token).toEqual('tokenGenerico');
      expect(state.profile.cpf).toEqual('33333333333');
      expect(state.profile.phone).toEqual('987654321');
      expect(state.profile.isPresident).toEqual(false);
      expect(state.profile.segment).toEqual('Pais de alunos');
      expect(state.profile.CAE_Type).toEqual('Estadual');
      expect(state.profile.CAE_UF).toEqual('DF - Distrito Federal');
      expect(state.profile.CAE_municipalDistrict).toEqual('Brasília');
      expect(state.profile.CAE).toEqual('DF');
    };

    const wrapper = shallow(
      <RegisterScreen
        asyncRegisterCounselor={asyncRegisterCounselor}
      />,
    );
    wrapper.setState({
      nuvemCode: 1,
      email: 'Conselheiro@email.com',
      name: 'Conselheiro',
      userName: 'Conselheiro@email.com',
      password: 'senha',
      token: 'tokenGenerico',
      profile: {
        cpf: '33333333333',
        phone: '987654321',
        isPresident: false,
        segment: 'Pais de alunos',
        CAE_Type: 'Estadual',
        CAE_UF: 'DF - Distrito Federal',
        CAE_municipalDistrict: 'Brasília',
        CAE: 'DF',
      },
    });
    const touch = wrapper.findWhere(c => c.key() === 'userCreation');
    expect(touch.length).toEqual(1);
    touch.simulate('press');
  });
});
