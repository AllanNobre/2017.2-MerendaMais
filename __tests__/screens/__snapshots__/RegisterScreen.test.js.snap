// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RegisterScreen renders as expected 1`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Header
    backButton={false}
    subTitle=""
    title="Merenda +"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
         CPF
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "gray",
        "borderRadius": 7,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginHorizontal": 15,
        "padding": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="black"
      name="user-circle"
      size={32}
      style={
        Object {
          "marginRight": 10,
        }
      }
    />
    <TextInput
      keyboardType="numeric"
      maxLength={11}
      onChangeText={[Function]}
      placeholder="Digite o seu CPF"
      placeholderTextColor="#95a5a6"
      returnKeyLabel="next"
      underlineColorAndroid="transparent"
      value=""
      width={280}
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
         Nome
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "gray",
        "borderRadius": 7,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginHorizontal": 15,
        "padding": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="black"
      name="face"
      size={32}
      style={
        Object {
          "marginRight": 10,
        }
      }
    />
    <TextInput
      keyboardType="default"
      maxLength={60}
      onChangeText={[Function]}
      placeholder="Digite o seu nome completo"
      placeholderTextColor="#95a5a6"
      returnKeyLabel="next"
      underlineColorAndroid="transparent"
      value=""
      width={280}
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
         Email
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "gray",
        "borderRadius": 7,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginHorizontal": 15,
        "padding": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="black"
      name="email"
      size={32}
      style={
        Object {
          "marginRight": 10,
        }
      }
    />
    <TextInput
      keyboardType="email-address"
      maxLength={50}
      onChangeText={[Function]}
      placeholder="Digite o seu email"
      placeholderTextColor="#95a5a6"
      returnKeyLabel="next"
      underlineColorAndroid="transparent"
      value=""
      width={280}
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
         Telefone
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "gray",
        "borderRadius": 7,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginHorizontal": 15,
        "padding": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="black"
      name="phone"
      size={32}
      style={
        Object {
          "marginRight": 10,
        }
      }
    />
    <TextInput
      keyboardType="phone-pad"
      maxLength={11}
      onChangeText={[Function]}
      placeholder="Digite o seu telefone"
      placeholderTextColor="#95a5a6"
      returnKeyLabel="next"
      underlineColorAndroid="transparent"
      value=""
      width={280}
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
         Cargo
  </Text>
  <View
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 7,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginHorizontal": 15,
      }
    }
  >
    <Picker
      mode="dialog"
      onValueChange={[Function]}
      selectedValue=""
    >
      <PickerItem
        color="#95a5a6"
        label="Escolha seu cargo"
        value=""
      />
      <PickerItem
        label="Presidente"
        value={true}
      />
      <PickerItem
        label="Conselheiro"
        value={false}
      />
    </Picker>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
         Tipo de Conselheiro
  </Text>
  <View
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 7,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginHorizontal": 15,
      }
    }
  >
    <Picker
      mode="dialog"
      onValueChange={[Function]}
      selectedValue=""
    >
      <PickerItem
        color="#95a5a6"
        label="Escolha seu cargo"
        value=""
      />
      <PickerItem
        label="Titular"
        value="Titular"
      />
      <PickerItem
        label="Suplente"
        value="Suplente"
      />
    </Picker>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
         Segmento
  </Text>
  <View
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 7,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginHorizontal": 15,
      }
    }
  >
    <Picker
      mode="dialog"
      onValueChange={[Function]}
      selectedValue=""
    >
      <PickerItem
        color="#95a5a6"
        label="Escolha seu segmento"
        value=""
      />
      <PickerItem
        label="Poder executivo"
        value="Poder executivo"
      />
      <PickerItem
        label="Trabalhadores da educação"
        value="Trabalhadores da educação"
      />
      <PickerItem
        label="Pais de alunos"
        value="Pais de alunos"
      />
      <PickerItem
        label="Entidades civis"
        value="Entidades civis"
      />
    </Picker>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
         Tipo do CAE
  </Text>
  <View
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 7,
        "borderWidth": 1,
        "marginBottom": 10,
        "marginHorizontal": 15,
      }
    }
  >
    <Picker
      mode="dialog"
      onValueChange={[Function]}
      selectedValue=""
    >
      <PickerItem
        color="#95a5a6"
        label="Escolha o Tipo do seu CAE"
        value=""
      />
      <PickerItem
        label="Municipal"
        value="Municipal"
      />
      <PickerItem
        label="Estadual"
        value="Estadual"
      />
    </Picker>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
         CAE
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderColor": "gray",
        "borderRadius": 7,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 10,
        "marginHorizontal": 15,
        "padding": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="black"
      name="location-city"
      size={32}
      style={
        Object {
          "marginRight": 10,
        }
      }
    />
    <TextInput
      keyboardType="default"
      maxLength={40}
      onChangeText={[Function]}
      placeholder="Lista com o CAE do seu município/estado"
      placeholderTextColor="#95a5a6"
      returnKeyLabel="next"
      underlineColorAndroid="transparent"
      value=""
      width={280}
    />
  </View>
  <TouchableOpacity
    activeOpacity={0.7}
    key="userCreation"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#FF9500",
        "borderRadius": 7,
        "borderWidth": 1,
        "marginBottom": 20,
        "marginHorizontal": 15,
        "marginTop": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "textAlign": "center",
        }
      }
    >
      Concluir
    </Text>
  </TouchableOpacity>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#a9a9a9",
        "borderTopWidth": 1,
        "flex": 0.7,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.6}
      onPress={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        Já tem um cadastro?
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "blue",
            }
          }
        >
           Entrar
        </Text>
      </Text>
    </TouchableOpacity>
  </View>
</ScrollViewMock>
`;
